### PR TITLE
Feature/issues/190

### PR DESCRIPTION
**Pull request by atropnikov**

### CODE REVIEW

File: Extension/lib/pages/integration.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+$(document).ready(function () {
+
+    var DISABLE_INTEGRATION_TEXT = 'Отключить режим интеграции';
+    var ENABLE_INTEGRATION_TEXT = 'Включить режим интеграции';
+    var FIND_INTEGRATION_APP_TEXT = 'Найти приложение для интеграции';
+
+    var INTEGRATION_APP_SEARCH_TEXT = 'Пожалуйста, подождите идет поиск приложения для интеграции.';
+    var INTEGRATION_DISABLED_TEXT = 'Режим интеграции выключен';
+    var INTEGRATION_APP_FOUND_TEXT = 'Приложения для интеграции найдено. Работает ';
+    var INTEGRATION_APP_NOT_FOUND_TEXT = 'Приложение для интеграции не найдено';
```
Comment (Mizzick) [inline Extension/lib/pages/integration.js]: localisations?

Comment (atropnikov) [inline Extension/lib/pages/integration.js]: It's non final version. Like a draft, only for debugging.

Comment (atropnikov) [inline Extension/lib/pages/integration.js]: I mean integration.html ui page of course.

File: Extension/lib/adguard.js [hunk 2]
```diff
@@ -56,19 +56,38 @@ var adguard = (function () { // jshint ignore:line
...
-     * @returns {boolean}
-     */
-    var isModuleSupported = function (module) {
-        return module in adguard;
+    function notImplemented() {
+        return false;
+    }
+
+    var hitStatsModule = {
+        addRuleHit: notImplemented,
```
Comment (ameshkov) [inline Extension/lib/adguard.js]: What's that for?

File: Extension/lib/pages/integration.js (hunk 1 above)
Comment (ameshkov) [inline Extension/lib/pages/integration.js]: Anyway, default strings must be in english

File: Extension/lib/utils/user-settings.js [hunk 3]
```diff
@@ -190,6 +191,14 @@ adguard.settings = (function (adguard) {
         setProperty(settings.DEFAULT_WHITE_LIST_MODE, enabled);
     };
 
+    var isIntegrationDisabled = function () {
```
Comment (ameshkov) [inline Extension/lib/utils/user-settings.js]: That's weird to check if something is `disabled`.

It'd be better to have `isIntegrationEnabled`/`setIntegrationEnabled(value)` functions

File: Extension/lib/app-integration.js [hunk 4]
```diff
@@ -0,0 +1,602 @@
...
+
+        return [gen(), gen(), gen(), gen(), gen(), gen(), gen(), gen()].join('');
+    }
+
+    /**
+     * Get or generate authorization token.
+     * Token is generated only once and saved to local storage.
+     */
+    function getOrGenerateToken() {
+        var token = adguard.localStorage.getItem('integration-token');
```
Comment (ameshkov) [inline Extension/lib/app-integration.js]: Disabling integration in extension settings should clear this token.

File: Extension/lib/adguard.js (hunk 2 above)
Comment (atropnikov) [inline Extension/lib/adguard.js]: That's dummy methods for simple adguard api.

**Pull request closed without merging**